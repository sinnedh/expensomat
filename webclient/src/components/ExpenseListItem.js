import React from 'react';

export default (props) => (
    <li>
      <strong>{props.expense.description}</strong>:&nbsp;
      {props.expense.amount / 100} Euro
      paid by {props.expense.paid_by.map((m, i) => <span key={i}>{m.name}</span>)}
      &nbsp;for {props.expense.paid_for.map((m, i) => <span key={i}>{m.name}</span>)}
    </li>
)
