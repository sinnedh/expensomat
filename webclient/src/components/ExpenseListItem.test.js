import React from 'react';
import ReactDOM from 'react-dom';
import ExpenseListItem from './ExpenseListItem';

it('renders with all params', () => {
  const div = document.createElement('div');
  const expense = {
    paid_for: [{ name: "Dennis" }],
    paid_by: [{ name: "Micha" }, { name: "Achim" } ],
    description: "Pasta beim Italiener",
    amount: 800,
    paid_at: "2017-11-11 15:28",
  }
  ReactDOM.render(<ExpenseListItem expense={expense} />, div);
});
