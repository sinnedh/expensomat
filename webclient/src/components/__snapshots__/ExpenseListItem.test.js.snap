// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseListItem matches snapshot when canEditExpense=false 1`] = `
<li>
  2017-11-11
  :
   
  <strong>
    Pasta beim Italiener
  </strong>
   
  8 Euro
   paid by 
  <span
    key="0"
  >
    Micha
  </span>
  <span
    key="1"
  >
    Achim
  </span>
   for 
  <span
    key="0"
  >
    Dennis
  </span>
  <Link
    replace={false}
    to="/calculation/ABCD1234/expense/undefined"
  >
    Details
  </Link>
</li>
`;

exports[`ExpenseListItem matches snapshot when canEditExpense=true 1`] = `
<li>
  <button
    onClick={[Function]}
  >
    X
  </button>
  2017-11-11
  :
   
  <strong>
    Pasta beim Italiener
  </strong>
   
  8 Euro
   paid by 
  <span
    key="0"
  >
    Micha
  </span>
  <span
    key="1"
  >
    Achim
  </span>
   for 
  <span
    key="0"
  >
    Dennis
  </span>
  <Link
    replace={false}
    to="/calculation/ABCD1234/expense/undefined"
  >
    Details
  </Link>
</li>
`;
